app/controllers/application_controller.rb

```diff
@@ -57,10 +57,10 @@ def restrict_iframes
   def enable_embedded_shopfront
     whitelist = Spree::Config[:embedded_shopfronts_whitelist]
     return unless Spree::Config[:enable_embedded_shopfronts] && whitelist.present?
-    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test? && !Rails.env.development?
+    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test? && !Rails.env.development? && !whitelist.include?(URI(request.referer).host)
```

Comment:
Did we consider extracting this logic from the application controller? This line is getting hairy over time (the last two commits touch it) and thus becoming harder to change. Besides, the embedded stuff adds an extra responsibility the controller. We should extract it from here as per the single responsibility principle.